packages/integration-sdk-runtime/src/execution/executeIntegration.ts

```diff
@@ -288,6 +290,17 @@ export async function executeWithContext<
         data: summary,
       });
 
+      if (resultsCallback != null) {
+        try {
+          await resultsCallback(integrationStepResults);
```

Comment:
The summary.json that we store locally is something like this:
```typescript
{
  integrationStepResults,
  metadata: {
    partialDatasets,
  },
}
```
I wonder if we should add the `metadata` object, same as the existing version?

Reply:
Maybe I was over-engineering, but I thought the `resultsCallback` should provide the raw results, and then the function is responsible for summarizing it or whatever.

But if we don't think that's useful I can just pass in the `summary` instead.
Or I can also have the function that stores in S3 in the managed runtime run the summarization function before it sends it to S3.

Reply:
I think the current version is fine, if we need to change it we can do it later, but IMO it's fine!